### PR TITLE
Rewrite the code to handle handwritten resources

### PR DESCRIPTION
**Pull request by zli82016**
1. Rewrite the logic to handle handwritten resources. Use Go code to replace `common~copy.yaml` and `common~compile.yaml` files.
2. Convert a couple of `.erb` handwritten files to `.tmpl` files.
3. Remove files that are not compiled to downstream.
    * mmv1/third_party/terraform/utils/provider_versionfive_upgrade_test.go.erb
    * mmv1/third_party/terraform/website/docs/r/compute_firewall_policy_rule.html.markdown

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/provider/template_data.go [hunk 1]
```diff
@@ -161,10 +161,6 @@ func (td *TemplateData) GenerateFile(filePath, templatePath string, input any, g
 	}
 
 	sourceByte := contents.Bytes()
-	// Replace import path based on version (beta/alpha)
```
Comment (zli82016) [inline mmv1/provider/template_data.go:164]: The function [importpath](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/resource.go.tmpl#L56C19-L56C29) is called in the templates files to decide the import path based on version. With `importpath`, the replacement is not needed.

**Comment by zli82016** [comment]
cc @NickElliot

File: mmv1/third_party/terraform/go.mod [hunk 2]
```diff
@@ -0,0 +1,110 @@
+module github.com/hashicorp/terraform-provider-google
```
Comment (c2thorn) [inline mmv1/third_party/terraform/go.mod:1]: being able to remove the templating on this file is great

File: mmv1/third_party/terraform/.goreleaser.yml.tmpl [hunk 3]
```diff
@@ -0,0 +1,83 @@
+archives:
+  - files:
+      # Include built binary and license files in archive
+      - 'LICENSE'
+    format: zip
+    name_template: '{{"{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}"}}'
```
Comment (c2thorn) [inline mmv1/third_party/terraform/.goreleaser.yml.tmpl:6]: wish there was a better way for these escapes... makes it very hard to read

**Review by c2thorn — APPROVED** [review]
LGTM

File: mmv1/third_party/terraform/.goreleaser.yml.tmpl (hunk 3 above)
Comment (NickElliot) [reply mmv1/third_party/terraform/.goreleaser.yml.tmpl:6]: They're all over the resource.go.tmpl too 🙃

Comment (zli82016) [reply mmv1/third_party/terraform/.goreleaser.yml.tmpl:6]: The function [Delims](https://medium.com/@etiennerouzeaud/change-default-delimiters-templating-with-template-delims-857938a0b661) can change the default delimiter. If we change the delimiter, we need to make the change in all of templates?

Comment (c2thorn) [reply mmv1/third_party/terraform/.goreleaser.yml.tmpl:6]: oh wow. Yeah we should think about doing that. Doesn't have to be now.

The downside would be we'd be deviating from standard go templating, and I don't have a pulse on how annoying that would be to contributors

**Pull request merged**